File: backend/internal/models/atlas_event.go
package models

import "github.com/lib/pq"

type HistoricalEra string

const (
	EraMossi        HistoricalEra = "mossi"
	EraBobo         HistoricalEra = "bobo"
	EraColonial     HistoricalEra = "colonial"
	EraIndependance HistoricalEra = "independance"
	EraSankara      HistoricalEra = "sankara"
)

type AtlasEvent struct {
	Base
	Era         HistoricalEra  `gorm:"not null" json:"era"`
	Year        int            `json:"year"`
	Title       string         `gorm:"not null" json:"title"`
	Subtitle    string         `json:"subtitle"`
	Description string         `json:"description"`
	Tags        pq.StringArray `gorm:"type:text[]" json:"tags"`
	ImageURL    string         `json:"imageUrl"`
	GradientCSS string         `json:"gradientCss"`
	SortOrder   int            `json:"sortOrder"`
}

File: mobile/lib/models/atlas_event.dart
class AtlasEvent {
  final int id;
  final String title;
  final String description;
  final int year;
  final String era;
  final String imageUrl;
  final double? lat;
  final double? lng;

  const AtlasEvent({
    required this.id,
    required this.title,
    this.description = '',
    this.year = 0,
    this.era = '',
    this.imageUrl = '',
    this.lat,
    this.lng,
  });

  factory AtlasEvent.fromJson(Map<String, dynamic> json) => AtlasEvent(
        id: json['id'] ?? json['ID'] ?? 0,
        title: json['title'] ?? '',
        description: json['description'] ?? '',
        year: json['year'] ?? 0,
        era: json['era'] ?? '',
        imageUrl: json['imageUrl'] ?? '',
        lat: (json['lat'] as num?)?.toDouble(),
        lng: (json['lng'] as num?)?.toDouble(),
      );
}
